js(docs): Update Tracing pages for stream mode (#17862)
getsentry/sentry-docs: docs/platforms/javascript/common/tracing/troubleshooting/index.mdx

sidebar_order: 9000
---

> If you need help managing transactions or spans, you can read more here. If you need additional help, you can ask on GitHub. Customers on a paid plan may also contact support.

Reviewer: ```suggestion
If you need help managing transactions or spans, start with this page. If you need additional help, you can ask on GitHub. Customers on a paid plan may also contact support.
```

`https://empowerplant.io/api/0/projects/ep/setup_form/?user_id=314159265358979323846264338327&tracking_id=EasyAsABC123OrSimpleAsDoReMi&product_name=PlantToHumanTranslator&product_id=1618033988749894848`

<PlatformContent includePath="performance/control-data-truncation" />

<PlatformSection notSupported={["javascript.effect", "javascript.cordova", "javascript.elysia"]}>
## Traces Miss Spans, High Memory Usage, or Data Loss After Crashes

If you're hitting the 1,000-span limit, experiencing high memory usage from long-running processes, or losing span data when your process crashes, consider enabling stream mode. Stream mode sends spans to Sentry in batches as they finish rather than holding them in memory until the transaction ends.

See <PlatformLink to="/tracing/new-spans/">New Spans</PlatformLink> for more information.

> ## `ignoreSpans` Rules No Longer Work After Migrating to Stream Mode

Reviewer: Do they no longer work, or does the behavior change?

Author: Updated this to "... No Longer Work As Expected After Migrating ..."

Reviewer: I guess I'm still not entirely clear what that means. Does it just not work, or will it still work but differently? We should be explicit.

Author: I updated the description in this section to be more specific.
The title is fitting imo though -- for some users it could not work at all, for others some spans get dropped correctly and some not -- it depends on their configuration

In <PlatformLink to="/tracing/new-spans">stream mode</PlatformLink>, `ignoreSpans` is evaluated at span start rather than at transaction end as in transaction mode. This means rules that match on names and attributes added or updated while a span is active may no longer match the intended spans. Review your `ignoreSpans` rules after migrating to make sure the names and attributes you're matching on are available when the span is created.

If you're auto-instrumenting and don't know what the initial name of a span is when it starts, enable SDK debug logging during development by setting <PlatformLink to="/configuration/options/#debug">`debug: true`</PlatformLink> when initializing the SDK.

> 

Reviewer 2: l: probably worth mentioning here as well

```suggestion

Additionally, make sure that you migrated any entries from `ignoreTransactions` to `ignoreSpans` as `ignoreTransactions` is not applied in stream mode.
```

Author: Thanks -- added!
